includes/class-wc-gateway-stripe.php

```diff
@@ -1015,4 +997,35 @@ public function get_intent_object( $intent_id = '' ) {
 
 		return $intent_object;
 	}
+
+	/**
+	 * Prepares and retrieves a new PaymentIntent from Stripe's API.
+	 *
+	 * @see https://stripe.com/docs/payments/payment-intents/quickstart
+	 * @return array An array with the intent `id` and `client_secret`.
+	 */
+	public function get_intent_and_secret() {
+		$total = WC()->cart->total;
+
+		if ( empty( WC()->cart->get_cart_contents() ) ) {
+			$message = __( 'Your cart is empty.', 'woocommerce-gateway-stripe' );
+			throw new WC_Stripe_Exception( 'empty_cart', $message );
+		}
+
+		$request = array(
+			'amount'               => WC_Stripe_Helper::get_stripe_amount( $total ),
+			'capture_method'       => 'manual',
+			'currency'             => get_woocommerce_currency(),
+			'allowed_source_types' => array(
+				'card',
+			),
+		);
+
+		$intent = WC_Stripe_API::request( $request, 'payment_intents' );
+
+		return array(
+			'id'            => $intent->id,
```

Comment:
I don't see the `id` property used anywhere. Shouldn't we keep a reference to the intent somewhere in the server?

Reply:
More than that, we should use webhooks. This is a bigger task, so I created an issue (#)[https://github.com/woocommerce/woocommerce-gateway-stripe/issues/804], which deserves a separate PR.